### ISSUE TITLE
[BUG] Converting to cv::Mat causes segmentation fault

### ISSUE THREAD

**Issue by abrock** [AcademySoftwareFoundation/OpenImageIO#3868]
**Describe the bug**

I'm trying to use OIIO for reading some files OpenCV can't read.
I made a minimal example: https://github.com/abrock/oiio-bugreport
The program is supposed to read an image, convert it to cv::Mat and print its dimensions but instead I get a segfault with this backtrace:

1 __memmove_avx_unaligned_erms                                                                                                                                                        memmove-vec-unaligned-erms.S 396 0x7ffff64d43d5 
2 OpenImageIO_v2_2::copy_image(int, int, int, int, void const *, long, long, long, long, void *, long, long, long)                                                                                                     0x7ffff7bb0bcd 
3 OpenImageIO_v2_2::parallel_convert_image(int, int, int, int, void const *, OpenImageIO_v2_2::TypeDesc, long, long, long, void *, OpenImageIO_v2_2::TypeDesc, long, long, long, int)                                  0x7ffff7bb35f8 
4 OpenImageIO_v2_2::ImageBufAlgo::to_OpenCV(cv::Mat&, OpenImageIO_v2_2::ImageBuf const&, OpenImageIO_v2_2::ROI, int)                                                                                                   0x7ffff7aff1bf 
5 main                                                                                                                                                                                main.cpp                     26  0x55555557255d 

This is the relevant code snippet without debug output:

    ImageBuf in(fn);
    bool const read_success = in.read();
    cv::Mat result;
    ImageBufAlgo::to_OpenCV(result, in);
    println("Size: [{}, {}]", result.cols, result.rows);

**To Reproduce**

git clone https://github.com/abrock/oiio-bugreport
cd oiio-bugreport/
cmake .
make
./bugreport 8bit.tif

-> segmentation fault.

**Expected behavior**

No segmentation fault

**Evidence**

https://github.com/abrock/oiio-bugreport

**Platform information:**
- OIIO branch/version: OpenImageIO 2.2.10
- OS: Debian bullseye
- C++ compiler: gcc version 10.2.1 20210110 (Debian 10.2.1-6)

**Comment by lgritz** [comment]
Do you know what version of OpenCV you built OIIO against?

Do you build OIIO yourself, or can you? I'm wondering if you can quickly try to build against the current release -- we're in the 2.4 series now (and developing 2.5), so we should try to quickly find out if this is a current bug or if it's something that was fixed long ago.

**Comment by enziop** [comment]
for me it does not segfault but seems to output the wrong dimensions.
It seems this line does not even return a success:
`ImageBufAlgo::to_OpenCV(result, in);`
but 
`OIIO:geterror()`
is also empty, so i don't know what happens.
adding the following line right after loading, shows that the ImageBuf has the right dimensions already:
`println("oiio size: [{}, {}], nchannels: {}", in.spec().width, in.spec().height, in.spec().nchannels);`
so maybe the `to_OpenCV` function can not handle an uninitialized `cv::Mat` ?
```
./bugreport 8bit.tif
Opening image 8bit.tif
Reading image 8bit.tif
Read success: true
oiio size: [100, 100], nchannels: 1
Converting image 8bit.tif to cv::Mat
Size: [0, 0]
```

OpenCV version 4.6.0
OpenImageIO version 2.4.10.0

**Comment by abrock** [comment]
> Do you know what version of OpenCV you built OIIO against?

4.7.0, compiled myself because I like the QT GUI elements and they are not available in the Debian package for OpenCV.

> Do you build OIIO yourself, or can you?

I did that (cfaccc1609266c91aabb03470287a3ada7178cfc), and made sure that I'm including and linking the correct version. This is the compile output (`ninja -v`):

    /usr/bin/g++  -I/home/alexander/sw/oiio/dist/include -isystem /usr/local/include/opencv4 -g -MD -MT CMakeFiles/bugreport.dir/main.o -MF CMakeFiles/bugreport.dir/main.o.d -o CMakeFiles/bugreport.dir/main.o -c /home/alexander/sw/oiio-bugreport/main.cpp
    [2/2] : && /usr/bin/g++ -g -rdynamic CMakeFiles/bugreport.dir/main.o -o bugreport -L/home/alexander/sw/oiio/dist/lib   -L/usr/local/lib   -L/home/alexander/sw/oiio/ext/dist/lib -Wl,-rpath,/home/alexander/sw/oiio/dist/lib:/usr/local/lib:/home/alexander/sw/oiio/ext/dist/lib  /usr/local/lib/libopencv_core.so.4.7.0  -lOpenImageIO  -lOpenImageIO_Util  /home/alexander/sw/oiio/ext/dist/lib/libPtex.so.2.4

The issue remains exactly the same.

> adding the following line right after loading, shows that the ImageBuf has the right dimensions already:

I added that to my repo and included printing the `channel_bytes()` value. I also tried initializing the `cv::Mat` before calling `to_OpenCV`, same result.

**Comment by lgritz** [comment]
I believe this was fixed by #4013
